daemon/graphdriver/devmapper/driver.go

```diff
@@ -122,12 +123,18 @@ func (d *Driver) GetMetadata(id string) (map[string]string, error) {
 // Cleanup unmounts a device.
 func (d *Driver) Cleanup() error {
 	err := d.DeviceSet.Shutdown(d.home)
+	umountErr := mount.RecursiveUnmount(d.home)
 
-	if err2 := mount.RecursiveUnmount(d.home); err == nil {
-		err = err2
+	// in case we have two errors, prefer the one from Shutdown()
+	if err != nil {
+		return err
 	}
 
-	return err
+	if umountErr != nil {
+		return errors.Wrap(umountErr, "Error unmounting "+d.home)
```

Comment:
`errors.Wrapf(umountErr, "error unmounting %s", d.home)`

Reply:
how is that better?

Reply:
errors are supposed to be lowercase for future wrapping in the caller

Reply:
I thought you were asking to change using `Wrap` and string concatenation to `Wrapf` and `%s` format, and overlooked the lowercase part.

Patch updated